catkin_tools/jobs/utils.py

```diff
@@ -66,8 +71,21 @@ def loadenv(logger, event_queue, job_env, package, context):
             quiet=True,
             cached=context.use_env_cache,
             strict=False)
-        job_env.update(resultspace_env)
-
+        for key, values_str in resultspace_env.items():
+            if key in updated_env:
+                updated_env[key] |= set(values_str.split(':'))
+            else:
+                updated_env[key] = set(values_str.split(':'))
+
+    # Now merge the updated environment into this job's environment, discarding duplicate
+    # values and ensuring that new ones are prepended.
+    for key, new_values_set in updated_env.items():
+        if key in job_env:
+            prepend_values = new_values_set - set(job_env[key].split(':'))
+            if prepend_values:
+                job_env[key] = ':'.join(prepend_values) + ':' + job_env[key]
```

Comment:
Another way could to drop the `if` and just do:

    job_env[key] = ':'.join(list(prepend_values) + job_env[key].split(':'))

Probably by keeping around the list since it's already split up a line above. The way it is seemed cleared to me though.